libnavigation-base/src/main/java/com/mapbox/navigation/base/route/NavigationRoute.kt

```diff
@@ -73,6 +78,25 @@ class NavigationRoute internal constructor(
             )
         }
 
+        /**
+         * Creates new instances of [NavigationRoute] based on the routes found in the [directionsResponseJson].
+         *
+         * Should not be called from UI thread. Contains serialisation and deserialisation under the hood.
```

Comment:
```suggestion
```

we use `Dispatchers.Default` for serialisation, so it should not be a problem to call the function from UI thread